rust/experimental/query_engine/kql-parser/src/scalar_expression.rs

```diff
@@ -73,6 +74,96 @@ pub(crate) fn parse_scalar_expression(
+        }
+        Rule::plus_token => {
+            let factor = inner_rules.next().unwrap();
+            (false, parse_arithmetic_factor(factor, state)?)
+        }
+        Rule::arithmetic_factor => (false, parse_arithmetic_factor(first, state)?),
+        _ => panic!("Unexpected rule in arithmetic_expression: {first}"),
+    };
+
+    // Apply unary minus if present
+    if has_unary_minus {
+        current_expr = ScalarExpression::Math(MathScalarExpression::Multiply(
+            BinaryMathmaticalScalarExpression::new(
+                query_location.clone(),
+                ScalarExpression::Static(StaticScalarExpression::Integer(
+                    IntegerScalarExpression::new(query_location.clone(), -1),
+                )),
+                current_expr,
+            ),
+        ));
+    }
+
+    // Process remaining addition/subtraction operations
+    while let Some(op_rule) = inner_rules.next() {
+        let right = parse_arithmetic_factor(inner_rules.next().unwrap(), state)?;
+
+        current_expr = match op_rule.as_rule() {
+            Rule::plus_token => ScalarExpression::Math(MathScalarExpression::Add(
+                BinaryMathmaticalScalarExpression::new(query_location.clone(), current_expr, right),
+            )),
+            Rule::minus_token => ScalarExpression::Math(MathScalarExpression::Subtract(
+                BinaryMathmaticalScalarExpression::new(query_location.clone(), current_expr, right),
+            )),
+            _ => panic!("Unexpected operator in arithmetic_expression: {op_rule}"),
+        };
+    }
+
+    Ok(current_expr)
+}
+
+fn parse_arithmetic_factor(
+    factor_rule: Pair<Rule>,
+    state: &ParserState,
+) -> Result<ScalarExpression, ParserError> {
+    let query_location = to_query_location(&factor_rule);
+    let mut inner_rules = factor_rule.into_inner();
+    let mut current_expr = parse_arithmetic_atom(inner_rules.next().unwrap(), state)?;
+
+    // Process multiplication/division/modulo operations
+    while let Some(op_rule) = inner_rules.next() {
+        let right = parse_arithmetic_atom(inner_rules.next().unwrap(), state)?;
+
+        current_expr = match op_rule.as_rule() {
+            Rule::multiply_token => ScalarExpression::Math(MathScalarExpression::Multiply(
+                BinaryMathmaticalScalarExpression::new(query_location.clone(), current_expr, right),
+            )),
+            Rule::divide_token => ScalarExpression::Math(MathScalarExpression::Divide(
+                BinaryMathmaticalScalarExpression::new(query_location.clone(), current_expr, right),
+            )),
+            Rule::modulo_token => ScalarExpression::Math(MathScalarExpression::Modulus(
+                BinaryMathmaticalScalarExpression::new(query_location.clone(), current_expr, right),
+            )),
+            _ => panic!("Unexpected operator in arithmetic_factor: {op_rule}"),
+        };
+    }
+
+    Ok(current_expr)
+}
+
+fn parse_arithmetic_atom(
```

Comment:
nit: Let's remove this method and just call into `parse_scalar_expression` directly.